accelerator-home-ui/src/api/TTSApi.js

```diff
@@ -0,0 +1,184 @@
+                    reject(err)
+                })
+        })
+    }
+    deactivate() {
+        return new Promise((resolve, reject) => {
+            thunder.Controller.deactivate({ callsign: callsign })
+                .then(() => {
+                    resolve(true)
+                })
+                .catch(err => {
+                    console.log('Error Deactivation', err)
+                    Metrics.error(Metrics.ErrorType.OTHER, errorName, `Error while Thunder Controller ${callsign} deactivate ${JSON.stringify(err)}`, false, null)
+                    reject(err)
+                })
+        })
+    }
+
+    enable(enabletts) {
+        return new Promise((resolve) => {
+          thunder
+            .call(callsign, 'enabletts', {
+              "enabletts": enabletts
+            })
+            .then(() => {
+              resolve(true)
+            })
+            .catch(err => {
+              console.error("TTSApi TextToSpeech enable error:", JSON.stringify(err, 3, null))
+              Metrics.error(Metrics.ErrorType.OTHER, "PluginError", "Error in Thunder TextToSpeech enable " + JSON.stringify(err), false, null)
+              resolve(false)
+            })
+        })
+      }
+
+    isEnabled() {
+        return new Promise((resolve) => {
+          thunder
+            .call(callsign, 'isttsenabled')
+            .then(result => {
+              resolve(result.isenabled)
+            })
+            .catch(err => {
+              console.error("TTSApi TextToSpeech isEnabled error:", JSON.stringify(err, 3, null))
+              Metrics.error(Metrics.ErrorType.OTHER, "PluginError", "Error in Thunder TextToSpeech isEnabled " + JSON.stringify(err), false, null)
+              resolve(false)
+            })
+        })
+    }
+
+      setTTSConfiguration(configuration) {
+        return new Promise((resolve) => {
+          thunder
+            .call(callsign, 'setttsconfiguration', configuration)
+            .then(() => {
+              resolve(true)
+            })
+            .catch(err => {
+              console.error("TTSApi TextToSpeech setTTSConfiguration error:", JSON.stringify(err, 3, null))
+              Metrics.error(Metrics.ErrorType.OTHER, "PluginError", "Error in Thunder TextToSpeech setTTSConfiguration " + JSON.stringify(err), false, null)
+              resolve(false)
+            })
+        })
+      }
+
+      getTTSConfiguration() {
+        return new Promise((resolve) => {
+          thunder
+            .call(callsign, 'getttsconfiguration')
+            .then(result => {
+              resolve(result)
+            })
+            .catch(err => {
+              console.error("TTSApi TextToSpeech getTTSConfiguration error:", JSON.stringify(err, 3, null))
+              Metrics.error(Metrics.ErrorType.OTHER, "PluginError", "Error in Thunder TextToSpeech getTTSConfiguration " + JSON.stringify(err), false, null)
+              resolve(false)
+            })
+        })
+      }
+
+    ttsSpeak() {
```

Comment:
Please use function names matching to method where ever it is possible to avoid confusion.
Eg: https://rdkcentral.github.io/entservices-apis/#/apis/TextToSpeechPlugin?id=speak 
Also expose parameters of the methods so that caller can specify what they want to do.